Restart Fixed - Snake Movement fixed

MiguelCLG opened the pull request:
- We needed to unsubscribe to events in order to be able to restart
- Snake now wraps up around the grid, teleports to the sides like the player

MiguelCLG committed:
Restart Fixed

- We needed to unsubscribe to events in order to be able to restart
- Snake doesn't lock itself if it has no path, but it does pick a random neighboor to go to, the issue here is that it can run out of neighboors and just eat itself :sweat_smile:

dirop1 reviewed (dismissed):
Assim esta melhor já com o restart, assim que puder faço o teste pra isto para mergerimos a branch de testes!!

MiguelCLG committed:
fix: a snake agora tem em conta a os lados da grid como se fosse continua

- A snake tem em conta os lados da grid
- Teleporta-se para o lado oposto quando o caminho é mais curto por aí.

dirop1 reviewed (dismissed):
Makes sense, that we have to unsubscribe. Don't know how we didnt think of that b4. Glad you fixed it.

MiguelCLG committed:
removed unused function
